libp2p/discovery/discoverer_interface.py

```diff
@@ -0,0 +1,17 @@
+from abc import ABC, abstractmethod
+# pylint: disable=too-few-public-methods
+
+
+class IDiscoverer(ABC):
+
+    def __init__(self):
+        pass
+
+    @abstractmethod
+    def find_peers(self, service):
+        """
+        Find peers on the networking providing a particular service
+        :param service: service that peers must provide
+        :return: peerstore containing found peers on the network
```

Comment:
If we look at the return value of the go implementation: https://github.com/libp2p/go-libp2p-discovery/blob/master/interface.go#L19
`find_peers` could return a generator or `PeerInfo` instead of a `PeerStore`
That would prevent to load too much in memory in case there are a lot of peers to be discovered

Reply:
Resolved in new commit